spec/support/match_contents_matcher.rb

```diff
@@ -1,7 +1,18 @@
 RSpec::Matchers.define :match_contents do |regexp|
   match do |filename|
-    contents = TestPaths.app_path.join(filename).read
-    contents =~ regexp
+    @filename = filename
+    @contents = TestPaths.app_path.join(filename).read
+
+    @contents =~ regexp
+  end
+
+  failure_message do
+    <<~MESSAGE
+      #{@filename} does not match the regex #{regexp}.
```

Comment:
`filename` is passed to `failure_message` if we want to get rid of the instance variable.

Reply:
Good call! Will do.